tests/integration/gwchecker.feature

```diff
@@ -0,0 +1,68 @@
+Feature: Gateway checker
+
+    Scenario: normal API request should still work with enabled GW checker
+        Given that I have enabled debug mode
+        And that I want to get all "Configs"
+        When I request "/config"
+        Then the response is JSON
+        And the response has a "count" property
+        And the type of the "count" property is "numeric"
+        And the guzzle status code should be 200
+
+    Scenario Outline: get debug information with enabled GW checker
+        Given that I have enabled debug mode
+        And that I want to get all "Configs"
+        When I request "/<endpoint>?gwcheck=<gwCheckValue>"
+        Then the response is JSON
+        And the response does not have a "count" property
+        And the "api.name" property equals "ushahidi:platform:gwcheck"
+        And the "api.version" property equals "0.1"
+        And the "data._GET.gwcheck" property equals "<gwCheckValue>"
+        And the "data._REQUEST.gwcheck" property equals "<gwCheckValue>"
+        And the "data._POST" property is empty
+        And the "data._SERVER.REQUEST_METHOD" property equals "GET"
+        And the "data._SERVER.QUERY_STRING" property equals "gwcheck=<gwCheckValue>"
+        And the "data._SERVER.HTTP_ACCEPT" property equals "application/json"
+        And the "data._SERVER.SCRIPT_NAME" property contains "/index.php"
+        And the "data._SERVER.REQUEST_URI" property equals "/api/v3/<endpoint>?gwcheck=<gwCheckValue>"
+        #the following line might fail when running tests on Apache
+        And the "data._SERVER.PATH_INFO" property equals "/api/v3/<endpoint>"
+        And the "data._SERVER.HTTP_USER_AGENT" property contains "GuzzleHttp"
+        And the "data._SERVER.DOCUMENT_ROOT" property contains "/platform"
+        And the "data._SERVER.SCRIPT_FILENAME" property contains "/platform/httpdocs/index.php"
```

Comment:
so this fails in codeship, where we appear to  have /httpdocs as doc root, not platform. Should we make it configurable maybe?
@tuxpiper @individual-it 

(sorry I didn't notice earlier 😬 )

Reply:
PHP 7.2.28 Development Server started 
 test/php-7.2 Listening on http://localhost:8000
 test/php-7.2 Document root is /var/www/httpdocs

Reply:
@rowasc that is strange, because it is the same in travis,
the root starts in `/home/travis/build/ushahidi/platform/httpdocs` the higher levels of the tree should not make any difference.

![image](https://user-images.githubusercontent.com/2425577/78318780-5a0b1f80-7585-11ea-81b0-64d77e458b0e.png)

I cannot access the codeship build to debug :-(

Reply:
https://github.com/ushahidi/platform/blob/develop/codeship-services.yml#L4
https://github.com/ushahidi/platform/blob/develop/docker/build.Dockerfile#L3 

I think this is what makes the difference for the codeship script from what I'm seeing . It's the same in the other dockerfiles we have setup ie the test one. 

Notice /platform is higher up than httpdocs in the example, but this test checks for /platform's existance, so if we don't have it, it wouldn't work 
ie: /var/www/httpdocs vs /var/www/*platform*/httpdocs

Reply:
ahh I see, overlooked that, there is no `platform` at all. Let me have a go at it

Reply:
an other try in #3911